### PR TITLE
feat(web): same-location pane splits in the right and bottom docks

### LINKED ISSUE #2486: Allow same-location pane splits within the right and bottom docks
**Note:** drafted by Codex via investigation. Idea and decisions are yours.

Part of #268. Follows the shipped tabbed-dock work in #2437 / PR #2446.

## Summary

The web dashboard currently lets panes move between the `right` and `bottom` docks, and lets each dock hold one tab group, but it does not let a user split within the same dock location the way tmux or iTerm can. In practice, dragging within a dock only reorders tabs, and dragging to the dock body only appends into that same single group.

The layout model already anticipates multiple groups per dock, so this is not a storage-gap problem. It is a render + interaction gap: the code still collapses each dock back to one group and mounts exactly one `Dock` per location.

## Current behavior

The current persisted layout shape stores `right` and `bottom` as arrays of `PaneGroup`, and the comment explicitly says split groups within a dock were deferred to a sibling follow-up (`web/src/lib/paneLayout.ts:23-28`).

However, the implementation still hard-codes a single group per dock:

- `firstGroup()` returns only `layout[dock][0]`, and `dockTabs()` / `dockActive()` read only that first group (`web/src/lib/paneLayout.ts:52-63`).
- `ensureGroup()` and `pruneEmpty()` only create/remove `layout[dock][0]` (`web/src/lib/paneLayout.ts:82-91`).
- `addTab()`, `removeTab()`, `setActive()`, and `placeTab()` all mutate only that first group (`web/src/lib/paneLayout.ts:94-163`).
- `normalizeGroups()` explicitly collapses multiple stored groups back into one merged group: “Collapse to a single group for now; DnD will lift this cap.” (`web/src/lib/paneLayout.ts:268-282`).

The dashboard render tree matches that limitation:

- `AppContent` computes one flat visible tab list for `right` and one for `bottom` via `dockTabs(...)`, then mounts exactly one `Dock` on the right and one `BottomDock` below (`web/src/App.tsx:455-470`, `web/src/App.tsx:1454-1480`).
- `BottomDock` itself is just a single `Dock` wrapper with a height resizer, not a container for multiple bottom groups (`web/src/components/BottomDock.tsx:33-107`).
- `Dock` renders one tab strip and one active body for a location (`web/src/components/Dock.tsx:104-186`).

Drag and drop is also limited to “reorder within one strip” or “move to the other dock”:

- The DnD controller only tracks `tabsByDock: Record<DockLocation, string[]>` and resolves drops to `{ dock, index }`, not to a group within that dock (`web/src/components/PaneDndController.tsx:46-87`).
- `resolvePlacement()` computes an insertion index inside the destination dock’s visible tab list, and non-tab drops append to that dock’s single group (`web/src/components/paneDnd.ts:54-68`).
- Empty landing zones exist only for an entirely empty `right` or `bottom` dock, not for creating another group inside an already-occupied dock (`web/src/components/PaneDndController.tsx:140-178`).

## Expected behavior

A user should be able to split panes within the same location:

- In the `right` dock, split one pane from another into two side-by-side groups.
- In the `bottom` dock, split one pane from another into two stacked groups.
- Same-location splits should persist per session, just like the existing dock/tab layout.
- Closing or moving one pane should repair active-tab and empty-group state without collapsing unrelated groups or duplicating tabs.

This should feel like tmux/iTerm pane splitting, not just tab reordering.

## Why this matters

The current model forces users to choose between:

- tabbing panes together and only seeing one at a time, or
- moving one pane to the opposite dock.

That is a narrower layout vocabulary than terminal multiplexers already teach users. It blocks common layouts like:

- two terminals side-by-side in the right dock,
- diff plus terminal stacked within the bottom dock,
- a plugin pane beside a terminal in the same dock while preserving the other dock for something else.

## Likely implementation seam

This looks like the intended next step after `#2437`, because the persisted shape already stores `PaneGroup[]` per dock.

Likely work items:

- Stop collapsing multiple groups in `normalizeGroups()`.
- Replace `firstGroup()`-style helpers with group-aware operations.
- Promote DnD targets from `{ dock, index }` to something like `{ dock, groupIndex, index }`, with explicit “split before/after” drop zones inside an occupied dock.
- Render multiple sibling `Dock` groups inside the right and bottom dock containers, with the correct orientation per location.
- Preserve active-tab repair and plugin-tab visibility rules across group boundaries.
- Add regression coverage for same-location split creation, persistence, and cleanup.

## User stories / regression tests

1. As a dashboard user, when I drag the active right-dock pane onto another right-dock pane’s edge, then AoE creates a second pane group in the right dock instead of only reordering tabs in one strip.
   Test layer hint: live Playwright.
2. As a dashboard user, when I split a bottom-dock pane from another bottom-dock pane, then both groups stay visible in the bottom dock across reloads for that session.
   Test layer hint: live Playwright.
3. As a dashboard user, when I have same-location split groups in a dock and close or move one pane, then the remaining group keeps a valid active tab and the dock layout does not collapse or duplicate tabs unexpectedly.
   Test layer hint: Vitest for layout helpers plus live Playwright for end-to-end behavior.

## Adjacent work checked

- `#2437` is the shipped tabbed-dock issue, and its body explicitly called “dragging a tab out into its own group or another dock” out of scope.
- PR `#2446` says the same thing in the merged implementation notes: same-location splitting was left for a follow-up, and simultaneous side-by-side panes currently require using `right` plus `bottom` rather than an intra-dock split.
- I did not find an exact existing issue for allowing same-location split groups in `right` / `bottom`.

### PR DESCRIPTION
**Pull request by Seluj78**
**Note:** Claude handled the implementation and prose via back-and-forth. Idea, decisions, and everything else are mine. — @Seluj78

## Description

Lets a user split panes within the same dock, the way tmux or iTerm split panes, instead of only tabbing them together or moving one to the opposite dock. Dragging a pane onto another pane's body in the same dock now lifts it into a second group: the tall right column stacks groups top to bottom, the wide bottom strip places them side by side. The split persists per session across reloads.

The persisted layout shape already stored `right`/`bottom` as `PaneGroup[]` (left ready by #2437 / PR #2446), so this is a render + interaction change, not a storage migration. The model was the blocker: every op used only `group[0]` and `normalizeGroups()` actively merged stored groups back into one on load.

What changed:
- **Model (`paneLayout.ts`)**: `dockTabs` now flattens across all groups (so "is this open / which dock / close everything here" stays correct); `dockGroups`, `findTab`, and `isActiveTab` are group-aware; `placeTab` takes one unified `PlaceTarget` covering reorder, cross-dock, cross-group, and split, repairing the source group's active tab and pruning only the emptied group (and adjusting the target index when the source group is removed). `normalizeGroups` no longer merges.
- **DnD**: drop targets carry a group index and a `newGroup` flag. Dropping on a group's tab strip joins it; dropping on a body half (shown only while dragging, so it never steals clicks from a terminal/iframe body) splits into a new group before/after by pointer position.
- **Render**: each dock maps its groups to one `Dock` each via a new `DockGroups` container. Groups share space equally; per-group resize handles are deliberately deferred (noted with a `ponytail:` comment for a follow-up).

The design was pressure-tested with a two-model debate (the unified `placeTab` vs a separate `splitTab`, flatten `dockTabs` vs migrate callers, edge split zones vs body-half split zones, and the hidden-plugin-slot index mapping going per-group).

Closes #2486

## PR Type

- [x] New Feature
- [ ] Bug Fix
- [ ] Refactor
- [ ] Documentation
- [ ] Infrastructure / CI

## Checklist

- [x] New and existing tests pass
- [x] Documentation was updated where necessary 
- [ ] For UI changes: included screenshot or recording

## Test Coverage Analysis

**Web dashboard / structured view (Playwright user story)**
- [x] Added or updated a Playwright spec under `web/tests/` and updated `web/tests/coverage-matrix.json`

**TUI / CLI (e2e test)**
- [x] N/A: this PR doesn't change TUI rendering, a CLI subcommand, or session lifecycle

## How to test

- [ ] Open the dashboard on a desktop viewport with a session that has diff + terminal open in the right dock.
- [ ] Drag the terminal tab onto the lower half of the pane body; confirm it splits into a second group stacked below instead of just reordering the tab strip.
- [ ] Reload the page; confirm the two stacked groups are still there for that session.
- [ ] Drag a bottom-dock pane onto the side half of another bottom-dock pane's body; confirm it splits into a second group side by side.
- [ ] Close one group's last pane; confirm only that group disappears, the other keeps a valid active tab, and no tab is duplicated.
- [ ] Drag a tab from one group's strip onto another group's strip; confirm it merges (joins) rather than splitting.

## AI Usage

- [x] AI was used

**AI Model/Tool used:** Claude (via Claude Code, agent-of-empires `/aoe-implement` skill)

**Any Additional AI Details you'd like to share:**
Investigation, the debate-driven design, implementation, and tests were drafted by Claude under my direction. I made the design calls (unified placement op, body-half split zones, deferring resizers) and reviewed each commit.

- [x] I am an AI Agent filling out this form (check box if true)

This PR enables tmux/iTerm-style “same-location” pane splits in the web dashboard: users can drag a terminal tab onto another pane’s body within the same dock to create a new pane group (rather than only tabbing together or switching docks). The right dock renders the resulting groups side-by-side, and the bottom dock renders them stacked. The layout now persists per session across reloads.

Key updates include making the pane layout and drag-and-drop flow group-aware (supporting multiple groups per dock), refactoring the UI to render dock groups separately, and updating drop targets so split joins vs tab-strip joins are distinguished correctly. Closing or moving panes is handled more safely so the active tab stays valid and unrelated groups aren’t accidentally collapsed.

Tests were added/updated (including Playwright) to cover group-based splitting, drag/drop placement behavior, and persistence across reloads.

Potential follow-ups: verify edge-case interactions when rapidly dragging between multiple groups (and across docks) and consider adding more accessibility/keyboard coverage for the new split targets.

### CODE REVIEW

File: web/src/components/paneDnd.ts [hunk 1]
```diff
@@ -2,69 +2,114 @@ import { createContext, useContext } from "react";
...
+    return { dock: over.dock, group: 0, newGroup: true };
+  }
+  const grp = groupsByDock[over.dock].find((g) => g.group === over.group);
+  const base = (grp?.tabs ?? []).filter((id) => id !== draggedId);
+  if (over.type !== "pane-tab") return { dock: over.dock, group: over.group, index: base.length };
   const overIndex = base.indexOf(over.tabId);
-  if (overIndex < 0) return { dock: over.dock, index: base.length };
-  return { dock: over.dock, index: overIndex + (over.after ? 1 : 0) };
+  if (overIndex < 0) return { dock: over.dock, group: over.group, index: base.length };
+  return { dock: over.dock, group: over.group, index: overIndex + (over.after ? 1 : 0) };
```
[the remark this answers is not in the document]
Comment (Seluj78) [reply web/src/components/paneDnd.ts:120]: Fixed in https://github.com/agent-of-empires/agent-of-empires/pull/2501/commits/041326100cedcd2c366b8646635fc53942278673 (with a resolvePlacement test for the no-op).

File: web/src/components/__tests__/Dock.test.tsx [hunk 2]
```diff
@@ -79,4 +81,29 @@ describe("Dock", () => {
...
+          renderBody={body}
+          onActivate={vi.fn()}
+          onClose={vi.fn()}
+          onMove={vi.fn()}
+        />
+      </PaneDndStateContext.Provider>,
+    );
+    expect(document.querySelector('[data-testid="pane-split-right-0-before"]')).toBeTruthy();
+    expect(document.querySelector('[data-testid="pane-split-right-0-after"]')).toBeTruthy();
+  });
```
[the remark this answers is not in the document]
Comment (Seluj78) [reply web/src/components/__tests__/Dock.test.tsx:112]: Done in https://github.com/agent-of-empires/agent-of-empires/pull/2501/commits/868d471236eab8baad28be80a7ff623f736fbff8 (render baseline, assert absent, unmount, then assert present in drag state).

**Comment by Seluj78** [comment]
Addressed the CodeRabbit review (all 5 findings):

- **App.tsx active terminal (1188)**: prefer an already-active terminal before the first for the paired-shell focus, so multi-group layouts focus the live terminal. https://github.com/agent-of-empires/agent-of-empires/pull/2501/commits/1357afaf
- **bottom-dock E2E (pane-docking.spec.ts)**: added a bottom-dock split spec mirroring the right-dock one (drag-to-split, reload-persists, close-prunes). https://github.com/agent-of-empires/agent-of-empires/pull/2501/commits/cdccf6ff
- **DockGroups comment (26-29)**: dropped the stray prefix, states the equal-flex tradeoff plainly. https://github.com/agent-of-empires/agent-of-empires/pull/2501/commits/6da796df

The two inline threads (paneDnd no-op, Dock test) are fixed and resolved.
